pkgs/applications/networking/browsers/webmacs/default.nix

```diff
@@ -0,0 +1,83 @@
+{ pkgs
+, lib
+, python3Packages
+, fetchFromGitHub
+}:
+
+let py3pk = python3Packages; in
+
+pkgs.python3.pkgs.buildPythonApplication rec {
+  name = "webmacs-${version}";
+  version = "0.8";
+
+  src = fetchFromGitHub {
+    owner  = "parkouss";
+    repo   = "webmacs";
+    rev    = "6d92ae8c2bf82cce854f34cb3d762932fe7fdc8b";
+
+    fetchSubmodules = true;
+
+    # The sha256 value was obtained by running:
+    # nix-prefetch-git https://github.com/parkouss/webmacs --fetch-submodules --rev 6d92ae8c2bf82cce854f34cb3d762932fe7fdc8b
+    # where the --rev given to nix-prefetch-git is the same as
+    # the rev variable in this set, at the time this comment was written :-)
+
+    sha256 = "1hzb9341hybgrqcy1w20hshm6xaiby4wbjpjkigf4zq389407368";
+
+    # Careful (at the time of writing, 2019-07-01) forgetting the
+    # --fetch-submodules argument to nix-prefetch-git, does NOT result
+    # in a hash mismatch (as it should, because the hashes of the repo
+    # with/without submodules should not match) and then we get a
+    # build error later on, when the submodule directories turn out to
+    # be empty.
+
+  };
+
+  propagatedBuildInputs = [
+    py3pk.pyqt5_with_qtwebkit
+    py3pk.dateparser
+    py3pk.jinja2
+  ];
+
+  buildInputs = [
+    py3pk.pygments
+    pkgs.qt5.qtbase
+  ];
+
+  doCheck = false;
+
+  checkInputs = [
+    py3pk.pytest
+    py3pk.pytest-xvfb
+    py3pk.pytest-qt
+    pkgs.herbstluftwm
+
+    # The following are listed in test-requirements.txt but appear not
+    # to be needed at present:
+
+    # py3pk.pytest-mock
+    # py3pk.flake8
+  ];
+
+  checkPhase = "pytest";
+
+  meta = with lib; {
+    description = "Keyboard-based web browser with Emacs/conkeror heritage";
+    longDescription = ''
+      webmacs is yet another browser for keyboard-based web navigation.
+
+      It mainly targets emacs-like navigation, and started as a clone (in terms of
+      features) of conkeror.
+
+      Based on QtWebEngine and Python 3. Fully customizable in Python.
+    '';
+    homepage = https://webmacs.readthedocs.io/en/latest/;
+    changelog = https://github.com/parkouss/webmacs/blob/master/CHANGELOG.md;
+    license = licenses.gpl3;
+    maintainers = with maintainers; [ jacg ];
+    platforms = platforms.all;
+  };
+
+  preBuild = "CC=g++";
```

Comment:
This doesn't look right. Nix should normally handle setting up your C compiler

Reply:
Hmm, yes ... it needs to use a C++ rather than a C compiler. Any hints on how else to persuade it to do so?

Reply:
The build fails with gcc?

Reply:
Yup:

````
gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/build/source/vendor/bloom-filter-cpp -I/build/source/vendor/hashset-cpp -I/build/source/vendor/ad-block -I/nix/store/96p42644i76jqgdkxjgyl3f0c8n8k16j-python3-3.7.4/include/python3.7m -c /build/source/c/adblock.c -o build/temp.linux-x86_64-3.7/build/source/c/adblock.o -g0 -std=c++11
cc1: warning: command line option ‘-std=c++11’ is valid for C++/ObjC++ but not for C
/build/source/c/adblock.c:19:10: fatal error: iostream: No such file or directory
 #include <iostream>
          ^~~~~~~~~~
compilation terminated.
error: command 'gcc' failed with exit status 1
builder for '/nix/store/lb8cqs8ypcsb94mj85r7k1kv1s73r51b-webmacs-0.8.drv' failed with exit code 1
````

... which is to be expected, when compiling C++ programs with a C compiler (unless you explicitly tell it where to find the C++ standard libraries ... but that's kinda the point of g++: it takes care of those sorts of things).

Reply:
I feel it should be using $CXX but I don't know what you can do if it doesn't (apart from what you have done

Reply:
You could try putting `CC = "g++"` in the derivation. If that works I think it is a bit neater

Reply:
What *exactly* do you mean by this (I'm still a little lost in nix terminology)? If I try putting `CC = "g++";` at the same level as `pname = "webmacs;` then it fails in exactly the same way.

Reply:
that is what I meant but clearly it doesn't work

Reply:
leave it as is for now, It is probably me being picky about something that is actually ok

Reply:
Let me poke around a bit to try to understand why it's using `$CC` rather than `$CXX`, but if I don't learn anything in a reasonably short amount of time, then I'll leave it as is. In the meantime I'll push all the other changes you requested.

Reply:
Whatever the result of this, this should go a bit further up in the file and not be hidden down here